src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -187,12 +195,40 @@ public static AadIssuerValidator GetAadIssuerValidator(string aadAuthority)
             return GetAadIssuerValidator(aadAuthority, null);
         }
 
+        /// <summary>
+        /// Get the effective authority based on the token version.
+        /// </summary>
+        /// <param name="authority">The authority to query for metadata.</param>
+        /// <param name="securityToken">Received security token.</param>
+        /// <returns></returns>
+        public static string GetEffectiveAuthorityFromToken(string authority, SecurityToken securityToken)
```

Comment:
Not sure here is the best place for this method.

Reply:
This needs to be changed to non-static if it is going to reference AadAuthorityV1 and AadAuthorityV2. Since there are unit tests verifying the values of AadAuthorityV1 and AadAuthorityV2, this method should use them instead of setting effectiveAuthority separately, otherwise the unit tests should be modified.

Reply:
This method need not be public.

Reply:
This is being used by SAL. Makes sense to be public.

Reply:
I am not seeing the need for this method.
The AadIssuerValidator cannot depend on external runtimes performing logic.
It must run independently, the flag on TokenValidationParameters that it is OK to use LKG, needs to be honored.